sdk/servicebus/Azure.Messaging.ServiceBus/samples/TopicFilters/Program.cs

```diff
@@ -0,0 +1,242 @@
+            else if (!string.IsNullOrEmpty(fullyQualifiedNamespace))
+            {
+                var defaultAzureCredential = new DefaultAzureCredential();
+                s_client = new ServiceBusClient(fullyQualifiedNamespace, defaultAzureCredential);
+                s_adminClient = new ServiceBusAdministrationClient(fullyQualifiedNamespace, defaultAzureCredential);
+            }
+            else
+            {
+                throw new ArgumentException(
+                    "Either a fully qualified namespace or a connection string environment variable must be specified.");
+            }
+
+            Console.WriteLine($"Creating topic {TopicName}");
+            await s_adminClient.CreateTopicAsync(TopicName);
+
+            s_sender = s_client.CreateSender(TopicName);
+
+            // First Subscription is already created with default rule. Leave as is.
+            Console.WriteLine($"Creating subscription {NoFilterSubscriptionName}");
+            await s_adminClient.CreateSubscriptionAsync(TopicName, NoFilterSubscriptionName);
+
+            Console.WriteLine($"SubscriptionName: {NoFilterSubscriptionName}, Removing and re-adding Default Rule");
+            await s_adminClient.DeleteRuleAsync(TopicName, NoFilterSubscriptionName, RuleProperties.DefaultRuleName);
+            await s_adminClient.CreateRuleAsync(TopicName, NoFilterSubscriptionName,
+                new CreateRuleOptions(RuleProperties.DefaultRuleName, new TrueRuleFilter()));
+
+            // 2nd Subscription: Add SqlFilter on Subscription 2
+            // Delete Default Rule.
+            // Add the required SqlFilter Rule
+            // Note: Does not apply to this sample but if there are multiple rules configured for a 
+            // single subscription, then one message is delivered to the subscription when any of the 
+            // rule matches. If more than one rules match and if there is no `SqlRuleAction` set for the
+            // rule, then only one message will be delivered to the subscription. If more than one rules
+            // match and there is a `SqlRuleAction` specified for the rule, then one message per `SqlRuleAction`
+            // is delivered to the subscription.
+            Console.WriteLine($"Creating subscription {SqlFilterOnlySubscriptionName}");
+            await s_adminClient.CreateSubscriptionAsync(TopicName, SqlFilterOnlySubscriptionName);
+
+            Console.WriteLine($"SubscriptionName: {SqlFilterOnlySubscriptionName}, Removing Default Rule and Adding SqlFilter");
+            await s_adminClient.DeleteRuleAsync(TopicName, SqlFilterOnlySubscriptionName, RuleProperties.DefaultRuleName);
+            await s_adminClient.CreateRuleAsync(
+                TopicName,
+                SqlFilterOnlySubscriptionName,
+                new CreateRuleOptions {Name = "RedSqlRule", Filter = new SqlRuleFilter("Color = 'Red'")});
+
+
+            // 3rd Subscription: Add SqlFilter and SqlRuleAction on Subscription 3
```

Comment:
Same general thoughts as above.   This does a good job of demonstrating the mechanics but even with knowledge of the subject matter, it's not clear from the comments what we're filtering for and what makes it significant for the purposes of demonstration in this sample.

I realize that we're focusing on the "how" but this topic is esoteric enough that I wonder if those who already know the details of "what" and "why" aren't already well positioned to understand the "how" form documentation.   I think it would be beneficial if we added some additional context and details to assume less familiarity and knowledge of readers.

Reply:
I really think that what you are looking for should be covered by the service docs. I don't think we'd want to repeat the same thing using slightly different language in 4 repos across 4 languages. The goal here is to provide samples that can be linked from these service docs for users who want to see how the feature can be implemented.

Reply:
I've got nothing against just pointing back to the docs,  I'm just looking at it from the perspective of someone browsing our repository and seeing the sample without context.

Reply:
Well isn't that what the README is for 😄

Reply:
You'd think...